core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -117,4 +120,39 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
           }
         });
   }
+
+  /**
+   * Load a custom {@link FileIO} implementation.
+   * The implementation must have a no-arg constructor.
+   * If the class implements {@link org.apache.hadoop.conf.Configurable}, we will also set the Hadoop configuration.
+   * @param impl full class name of a custom FileIO implementation
+   * @param hadoopConf hadoop configuration
+   * @return FileIO class
+   * @throws IllegalArgumentException if class path not found or
+   *  right constructor not found or
+   *  the loaded class cannot be casted to the given interface type
+   */
+  public static FileIO loadCustomFileIO(String impl, Configuration hadoopConf) {
+    LOG.info("Loading custom FileIO implementation: {}", impl);
+    DynConstructors.Ctor<FileIO> ctor;
+    try {
+      ctor = DynConstructors.builder(FileIO.class)
+          .impl(impl)
+          .buildChecked();
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException(String.format(
+          "Cannot initialize FiloIO, please make sure %s has a no-arg constructor", impl), e);
+    }
+    try {
```

Comment:
Style: We like to have empty lines after control flow blocks like `for`, `while`, `if`, `else`, `try`, and `catch`. It helps keep the code at a readable density.